src/main/java/io/fabric8/jenkins/openshiftsync/BuildDecisionHandler.java

```diff
@@ -127,6 +144,23 @@ private void triggerBuildRequest(String buildConfigname, String namespace, Strin
       );
   }
 
+  private void cancelBuildRequest(String buildConfigname, String namespace, int number) {
```

Comment:
Move this to OpenShiftUtils and create a new versions of the method updateOpenShiftBuildPhase at https://github.com/fabric8io/jenkins-sync-plugin/blob/job-to-bc/src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java#L416-L422 that takes some extra params to set the trigger by and message params you want here.  Then have the original method with less params call the new one with more params, where you don't call editTriggerBy or withMessage if the params are empty or null